### PR TITLE
Added Change Java Version component doc

### PR DESCRIPTION
**Pull request by ftmzhrsckldr**

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -1166,4 +1166,9 @@ status = 301
 [[redirects]]
 from = "/testing-distribution/apple-devices"
 to = "/signing-identities/apple-devices"
-status = 301
\ No newline at end of file
+status = 301
+
+[[redirects]]
+from = "/workflows/common-workflow-steps/custom-script#how-to-change-java-version"
```
Comment (boztopuz) [inline netlify.toml:1172]: As I remember, Çelik sad that we can remove the example custom script in the faq of custom script. I test to redirection of this URL, I still go to old java version change cs example under CS Faq section.

If I am wrong, please let me know

**Review by boztopuz — CHANGES_REQUESTED** [review]

File: netlify.toml (hunk 1 above)
Comment (ftmzhrsckldr) [reply netlify.toml:1172]: > As I remember, Çelik sad that we can remove the example custom script in the faq of custom script.
right, I thought I removed it.

>  I test to redirection of this URL, I still go to old java version change cs example under CS Faq section.
It doesn't redirect even though I uninstall it.

Comment (boztopuz) [reply netlify.toml:1172]: Although the custom script example removed from Custom Scrip FAQ section, this link does not redirect to new component link. I tried with this link @csonuryilmaz 

https://deploy-preview-717--reverent-galileo-8ef035.netlify.app/workflows/common-workflow-steps/custom-script#how-to-change-java-version

**Review by boztopuz — COMMENTED** [review]
All review completed, only one thing that I mentioned is still ongoing. Netlify does not redirect to the new section.

File: netlify.toml (hunk 1 above)
Comment (csonuryilmaz) [reply netlify.toml:1172]: I had worked on the Netlify redirects about a year ago, so in order to identify the case here, I reworked the Netlify documents to remember the details.

Unfortunately, I saw that Netlify redirects have not been maintained well during this period, which can lead to weird cases later in the future.

👇 First of all, **anybody who will touch this file** should work on the below pages to understand the working principles that affect the redirection.

- https://docs.netlify.com/routing/redirects/
  - https://docs.netlify.com/routing/redirects/redirect-options/

When you work on these resources, you will see the reason behind the below "custom-script" redirection.

Screenshot 2024-10-27 at 09 40 57

I will not dive into those details and focus on the case here.

**TLDR;** `#` redirection is not supported in Netlify redirects.  You cannot use them in `from` section since the Netlify redirector does not see that part when a request comes it. (**[ref](https://answers.netlify.com/t/redirecting-from-invite-token-to-admin-invite-token/51661)**)

Let me explain the situation with the samples below:

Screenshot 2024-10-27 at 09 48 45

You can use `#` in the `to` section for a redirect rule (_it works_), but `from` section is useless.

We have two options here;

- We can keep that section in the document and add an info box there that explains that the relevant content is moved or changed and redirects the use to the new section.
- OR, we can work on Netlify's (or Docusaurus') other mechanisms to achieve the goal here, which can take more effort to discover other possibilities.

**Comment by csonuryilmaz** [comment]
> All review completed, only one thing that I mentioned is still ongoing. Netlify does not redirect to the new section. @csonuryilmaz

See the comment [here](https://github.com/appcircleio/appcircle-docusaurus/pull/717#discussion_r1817995559).

File: netlify.toml (hunk 1 above)
Comment (boztopuz) [reply netlify.toml:1172]: @csonuryilmaz As far as I understand, it makes more sense in the short term to keep the document for the time being and to provide guidance with an information message. If yes, @ftmzhrsckldr , could you please take back the sample code we provided under Custom script and reference the new component by redirecting to it?

File: docs/workflows/common-workflow-steps/custom-script.md [hunk 2]
```diff
@@ -35,72 +35,15 @@ Note that the **Script** area works according to the selected language variable.
...
-```txt
-openjdk version "17.0.7" 2023-04-18 LTS
-OpenJDK Runtime Environment Zulu17.42+19-CA (build 17.0.7+7-LTS)
-OpenJDK 64-Bit Server VM Zulu17.42+19-CA (build 17.0.7+7-LTS, mixed mode, sharing)
-```
-
-You can also switch to other pre-installed Java versions using the relevant environment variable as an argument in the `sdk` command. For more details about these environment variables, see the [Android Build Infrastructure](/infrastructure/android-build-infrastructure#java-version) page.
-
-:::
+- [Change Java Version - Source Code](https://github.com/appcircleio/appcircle-change-java-version-component)
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/custom-script.md:46]: @ftmzhrsckldr we can change the link with only github link. In this way, it can look like we give in the components source code.

for example: https://capture.dropbox.com/97kXqdO5GTdihygc

**Review by boztopuz — CHANGES_REQUESTED** [review]

File: docs/workflows/common-workflow-steps/custom-script.md (hunk 2 above)
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/custom-script.md:46]: @boztopuz I. did it this way on purpose. Because the other way is seem little complex but I think it is not necessary

Current:
Screenshot 2024-10-28 at 12 29 36

Feedback :
Screenshot 2024-10-28 at 12 26 42


It is not on the component page for that IMO we can applied different format.

Do you think it is "must to"?

Comment (boztopuz) [reply docs/workflows/common-workflow-steps/custom-script.md:46]: Since we do source code redirection for all components in this way, I thought it should fit the current structure. It is not mandatory, I don't think it will cause a problem in its current form. @ftmzhrsckldr

**Review by boztopuz — APPROVED** [review]

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+---
+title: Change Java Version
+description: Switch the current Java version to the selected one.
+tags: [change, java, version, java version, android java]
+---
+
+import Screenshot from '@site/src/components/Screenshot';
+
+# Change Java Version
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:9]:
```suggestion
# Select Java Version
```

This component name updated, we need to change doc title.

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+The **Change Java Version** step updates the JDK and Java version to the selected one during the build process.
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:11]:
```suggestion
The **Select Java Version** step updates the JDK and Java version to the selected one during the build process.
```

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+If you are unsure which Java version is required for your Android project, you can find it by following the steps in this FAQ:
+
+- [How can I check the Java version of my project?](#how-can-i-check-the-java-version-of-my-project)
+
+:::
+
+
+## Prerequisites
+
+There is no mandatory step for using this component.  However, if you have a step that necessitates changing the Java version (particularly build steps), the **Change Java Version** step should run before that step in the [workflow](/workflows).
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:34]:
```suggestion
There is no mandatory step for using this component.  However, if you have a step that necessitates changing the Java version (particularly build steps), the **Select Java Version** step should run before that step in the [workflow](/workflows).
```
And additionally, please change the prerequisites introduction sentence with the new template sentence that I change in this [PR](https://github.com/appcircleio/appcircle-docusaurus/pull/731).

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+For reference, you can see the workflow sequence in the image below:
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-change-java-version-1.0.png' />
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:38]: The component name has been updated, screen shot should be changed.

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+:::caution
+
+If your runner is self-hosted, ensure that the selected Java version is available in your environment. If not, you must install the required version before the **Change Java Version** step.
+
+:::
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:44]: Under this caution (line 40-44) we can add another caution box for this information that you are using in the introduction sentence. 

```suggestion
:::caution

if you have a step that necessitates changing the Java version (particularly build steps), the **Select Java Version** step should run before that step in the [workflow](/workflows)

:::
```

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Input Variables
+
+A required parameter for this step is outlined in the table below, along with its description.
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-change-java-version-2.0.png' />
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:51]: Screen shot should be changed. Comp name has been updated.

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+
+| Variable Name               | Description                                                                                                 | Status   |
+| --------------------------- | ----------------------------------------------------------------------------------------------------------- | -------- |
+| `$AC_SELECTED_JAVA_VERSION` | Specify the Java version to switch to. The default value is `17`. Options: `8`, `11`, `17`, `21`, and `23`. | Required |
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:56]:
```suggestion
| `$AC_SELECTED_JAVA_VERSION` | Specify the Java version to switch to. The default value is `17`. Options: `8`, `11`, `17`, `21`. | Required |
```

We need to remove java 23 version support, we don't support this yet

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+:::info
+
+If the selected Java version is not available on the runner, an error message will indicate that `this version is not found` and will display the available Java versions for your runner. Check available Java versions for cloud runner [here](/infrastructure/android-build-infrastructure#java-version).
+
+:::
+
+## Output Variables
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:64]: If this is not a specific description and we don't need to give this detail, we can remove it like I did in my [PR](https://github.com/appcircleio/appcircle-docusaurus/pull/731) template. Otherwise it can stay.

File: docs/workflows/common-workflow-steps/change-java-version.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+This step does not produce any visible output. However, if run successfully, it will update both the default Java version (`$JAVA_HOME`) and the system JDK version to the selected version.
+
+---
+
+To access the source code of this component, please use the following link:
+
+https://github.com/appcircleio/appcircle-change-java-version-component
+
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/change-java-version.md:73]: Please add the "---" here, It's a good way to separate the FAQ section from the main document and repo information.

File: docs/workflows/common-workflow-steps/custom-script.md [hunk 12]
```diff
@@ -35,72 +35,19 @@ Note that the **Script** area works according to the selected language variable.
...
-echo "OpenJDK 21" $JAVA_HOME_21_X64
-
-# Change JAVA_HOME to OPENJDK 17
-echo "JAVA_HOME=$JAVA_HOME_17_X64" >> $AC_ENV_FILE_PATH
-```
-
-Create a custom script like above and put it **above** your [Android Build](/workflows/android-specific-workflow-steps/android-build) step.
-
-<Screenshot url="https://cdn.appcircle.io/docs/assets/workflow-android-change-java-workflow.png" />
+To switch JDK versions, you can now use the dedicated **Change Java Version** component, so there is no need to use **Custom Script** for this task. For further details on this component, refer to the documentation:
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/custom-script.md:44]:
```suggestion
To switch JDK versions, you can now use the dedicated [**Select Java Version**](/workflows/common-workflow-steps/change-java-version) component, so there is no need to use **Custom Script** for this task. For further details on this component, refer to the documentation:
```

File: docs/workflows/common-workflow-steps/custom-script.md [hunk 13]
```diff
@@ -35,72 +35,19 @@ Note that the **Script** area works according to the selected language variable.
...
-:::caution
-
-Please be aware that this custom script affects any step that comes after.
-
-Therefore, you should use this step as a standalone step instead of as part of any custom script.
-
-:::
-
-:::tip
+- [Change Java Version](/workflows/common-workflow-steps/change-java-version)
```
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/common-workflow-steps/custom-script.md:46]: Link should be changed.

File: docs/workflows/common-workflow-steps/custom-script.md [hunk 14]
```diff
@@ -35,72 +35,19 @@ Note that the **Script** area works according to the selected language variable.
...
-```txt
-openjdk version "17.0.7" 2023-04-18 LTS
-OpenJDK Runtime Environment Zulu17.42+19-CA (build 17.0.7+7-LTS)
-OpenJDK 64-Bit Server VM Zulu17.42+19-CA (build 17.0.7+7-LTS, mixed mode, sharing)
-```
-
-You can also switch to other pre-installed Java versions using the relevant environment variable as an argument in the `sdk` command. For more details about these environment variables, see the [Android Build Infrastructure](/infrastructure/android-build-infrastructure#java-version) page.
-
-:::
+- [Change Java Version - Source Code](https://github.com/appcircleio/appcircle-change-java-version-component)
```
Comment (boztopuz) [inline docs/workflows/common-workflow-steps/custom-script.md:50]:
```suggestion
- [Select Java Version - Source Code](https://github.com/appcircleio/appcircle-change-java-version-component)
```

**Review by boztopuz — CHANGES_REQUESTED** [review]

File: docs/workflows/common-workflow-steps/change-java-version.md (hunk 6 above)
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/change-java-version.md:38]: Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.

File: docs/workflows/common-workflow-steps/change-java-version.md (hunk 9 above)
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/change-java-version.md:56]: Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.

File: docs/workflows/common-workflow-steps/change-java-version.md (hunk 11 above)
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/change-java-version.md:73]: Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.

File: docs/workflows/common-workflow-steps/custom-script.md (hunk 12 above)
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/custom-script.md:44]: Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.

File: docs/workflows/common-workflow-steps/custom-script.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (ftmzhrsckldr) [reply docs/workflows/common-workflow-steps/custom-script.md:46]: Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.

**Review by boztopuz — APPROVED** [review]
